src/Microsoft.Deployment.DotNet.Releases/src/ReleaseVersion.cs

```diff
@@ -568,5 +547,376 @@ internal static int CompareNumericIdentifier(string a, string b)
+        /// <param name="input">The string to check.</param>
+        /// <returns><see langword="true"/> if the string is a prerelease identifier; <see langword="false"> otherwise.</see></returns>
+        internal static bool IsPrereleaseIdentifier(string input)
+        {
+            return IsNumericIdentifier(input) || IsAlphaNumericIdentifier(input);
+        }
+
+
+        /// <summary>
+        /// Tries to convert the string representation of a version to an equivalent <see cref="ReleaseVersion"/> object,
+        /// and returns a value indicating whether the conversion succeeded. 
+        /// </summary>
+        /// <param name="input">A string that contains a version to convert.</param>
+        /// <param name="version">Contains the <see cref="ReleaseVersion"/> equivalent of the value in <paramref name="input"/> if
+        /// successful; otherwise <see langword="null"/>.</param>
+        /// <param name="throwOnFailure">A boolean indicating whether an exception should be raised if the conversion failed.</param>
+        /// <returns><see langword="true"/> if the string could be converted; <see langword="false"> otherwise.</see> </returns>
+        /// <exception cref="FormatException">If the string is an invalid semantic version.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">If the major, minor, or patch version exceeds <see cref="int.MaxValue"/>.</exception>
+        internal static bool TryParse(string input, bool throwOnFailure, out ReleaseVersion version)
+        {
+            version = null;
+
+            string build = null;
+            string prerelease = null;
+            string versionCore;
+
+            if (input == null)
+            {
+                if (throwOnFailure)
+                {
+                    throw new ArgumentNullException(nameof(input));
+                }
+
+                return false;
+            }
+
+            int buildSeparatorIndex = input.IndexOf('+');
+            int prereleaseSeparatorIndex = input.IndexOf('-');
+
+            if (buildSeparatorIndex == -1)
+            {
+                if (prereleaseSeparatorIndex == -1)
+                {
+                    // <valid semver> ::= <version core>
+                    versionCore = input;
+                }
+                else
+                {
+                    // <valid semver> ::= <version core> "-" <pre-release>
+                    if (prereleaseSeparatorIndex + 1 > input.Length)
+                    {
+                        if (throwOnFailure)
+                        {
+                            throw new FormatException(ReleasesResources.PrereleaseCannotBeEmpty);
+                        }
+
+                        return false;
+                    }
+
+                    prerelease = input.Substring(prereleaseSeparatorIndex + 1);
+                    versionCore = input.Substring(0, prereleaseSeparatorIndex);
+                }
+            }
+            else
```

Comment:
Consider utilizing an `else if` to reduce nesting

Reply:
Fixed